--- ./CMakeLists.txt~	2023-11-24 11:37:04.000000000 +0000
+++ ./CMakeLists.txt	2023-11-24 11:49:11.814012412 +0000
@@ -466,7 +466,7 @@
 function(unpack_db db_bzip2_file)
     set(KERNELS_DIR "${CMAKE_SOURCE_DIR}/src/kernels")
     STRING(REPLACE ".bz2" "" db_file "${db_bzip2_file}")
-    find_program(UNZIPPER lbunzip2 bunzip2)
+    find_program(UNZIPPER bunzip2 PATH /usr/bin)
     if(EXISTS "${db_file}")
         message(STATUS "WARNING: ${db_file} already exists and will be overwritten")
     endif()
