--- ./cmake/OCL.cmake~	2023-01-24 18:52:40.000000000 +0000
+++ ./cmake/OCL.cmake	2023-01-24 19:09:49.534991022 +0000
@@ -135,7 +135,7 @@
     # Extra link step with internalize
     COMMAND $<TARGET_FILE:llvm-link> -internalize -only-needed "${name}.link0${LIB_SUFFIX}"
       -o "${OUT_NAME}${LIB_SUFFIX}" ${internal_link_libs}
-    COMMAND $<TARGET_FILE:opt> -strip
+    COMMAND $<TARGET_FILE:opt> -passes=strip
       -o "${OUT_NAME}${STRIP_SUFFIX}" "${OUT_NAME}${LIB_SUFFIX}"
     COMMAND "${PREPARE_BUILTINS}"
       -o ${OUTPUT_BC_LIB} "${OUT_NAME}${STRIP_SUFFIX}"
