--- ./src/include/msccl/msccl_struct.h~	2023-10-06 21:05:52.000000000 +0100
+++ ./src/include/msccl/msccl_struct.h	2023-11-24 15:28:47.330693899 +0000
@@ -64,7 +64,7 @@
   int16_t channelId; // associated channel. -1 indicates a thread block with only local copies
 }; // 5384 bytes
 
-static_assert(sizeof(struct mscclThreadBlock) % sizeof(uint64_t) == 0, "Sanity check: sizeof(struct mscclThreadBlock) \% sizeof(uint64_t) != 0");
+static_assert(sizeof(struct mscclThreadBlock) % sizeof(uint64_t) == 0, "Sanity check: sizeof(struct mscclThreadBlock) \\% sizeof(uint64_t) != 0");
 
 struct mscclFlag {
   uint64_t flag;
