--- ./libomptarget/plugins/amdgpu/impl/impl.cpp~	2023-02-01 16:12:58.000000000 +0000
+++ ./libomptarget/plugins/amdgpu/impl/impl.cpp	2023-02-01 16:15:39.978466432 +0000
@@ -22,7 +22,7 @@
   err = hsa_amd_pointer_info(ptr, &info, nullptr, nullptr, nullptr);
 
   if (err != HSA_STATUS_SUCCESS)
-    DP("Error when getting pointer info\n");
+    DP("Error when getting pointer info\n")
   else
     is_locked = (info.type == HSA_EXT_POINTER_TYPE_LOCKED);
 
