--- ./libomptarget/CMakeLists.txt~	2023-02-01 16:50:27.000000000 +0000
+++ ./libomptarget/CMakeLists.txt	2023-02-01 16:51:12.026980017 +0000
@@ -161,10 +161,14 @@
 # lot of header definitions that get compiled with every offload compilation.
 add_subdirectory(libm)
 
-add_subdirectory(hostrpc)
+if(LIBOMPTARGET_BUILD_AMDGPU_PLUGIN)
+  add_subdirectory(hostrpc)
+endif()
 add_subdirectory(tools)
 set(PREP_TOOL $<TARGET_FILE:prep-libomptarget-bc>)
-add_subdirectory(deviceRTLs)
+if(LIBOMPTARGET_BUILD_AMDGPU_PLUGIN)
+  add_subdirectory(deviceRTLs)
+endif()
 add_subdirectory(DeviceRTL)
 
 # Add tests.
