--- a/onnxruntime/core/providers/rocm/math/softmax_ck.cuh	2023-11-18 04:58:21.000000000 +0000
+++ b/onnxruntime/core/providers/rocm/math/softmax_ck.cuh	2024-01-25 02:25:46.077043361 +0000
@@ -34,7 +34,7 @@
   using OutDataType = typename CKDataTypeAdaptor<OutputT>::type;
   using AccDataType = typename CKDataTypeAdaptor<AccT>::type;
   using DeviceSoftmax = ck::tensor_operation::device::
-      DeviceSoftmax<InDataType, AccDataType, OutDataType, Nop, Nop, Rank>;
+      DeviceSoftmax<InDataType, AccDataType, OutDataType, Nop, Nop, Rank, NumReduceDim>;
   using InstanceFactory = ck::tensor_operation::device::instance::DeviceOperationInstanceFactory<DeviceSoftmax>;
 
   std::vector<std::pair<std::string, tunable::Op<SoftmaxParams<InputT, OutputT>>>> ret;
@@ -49,9 +49,6 @@
       TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(
           params->is_log_softmax,
           impl->GetTypeString(), " does not support log softmax");
-      TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(
-          impl->GetRank() != Rank || impl->GetNumReduceDim() != NumReduceDim,
-          impl->GetTypeString(), " does not support current Rank or NumReduceDim ", params->Signature());
 
       std::vector<ck::index_t> in_lengths{1, 1, params->batch_count, params->softmax_elements};
       std::vector<ck::index_t> in_strides{params->batch_count * params->input_stride, params->batch_count * params->input_stride, params->input_stride, 1};
