--- ./onnxruntime/core/graph/graph.cc~	2024-01-31 00:51:05.000000000 +0000
+++ ./onnxruntime/core/graph/graph.cc	2024-02-09 14:12:53.441790691 +0000
@@ -9,6 +9,7 @@
 #include <numeric>
 #include <stack>
 #include <queue>
+#include <optional>
 
 #include "core/common/common.h"
 #include "core/common/gsl.h"
--- ./include/onnxruntime/core/graph/graph.h~	2024-01-31 00:51:05.000000000 +0000
+++ ./include/onnxruntime/core/graph/graph.h	2024-02-09 14:17:19.557631273 +0000
@@ -10,6 +10,7 @@
 #include <type_traits>
 #include <unordered_map>
 #include <unordered_set>
+#include <optional>
 
 #ifdef _WIN32
 #pragma warning(push)
