--- src/core/runtime/amd_blit_sdma.cpp~
+++ src/core/runtime/amd_blit_sdma.cpp
@@ -465,10 +465,11 @@
                       reinterpret_cast<uint32_t*>(out_signal.signal_.event_mailbox_ptr),
                       static_cast<uint32_t>(out_signal.signal_.event_id));
     command_addr += fence_command_size_;
+    command_addr += trap_command_size_;
     bytes_written_[wrapped_index] = post_bytes;
     wrapped_index += fence_command_size_;
 
     BuildTrapCommand(command_addr, out_signal.signal_.event_id);
     bytes_written_[wrapped_index] = post_bytes;
     wrapped_index += trap_command_size_;
   }
@@ -480,7 +481,7 @@
   if (pad_size) {
     memset(command_addr, 0, pad_size);
     uint32_t *dword_command_addr = reinterpret_cast<uint32_t*>(command_addr);
-    dword_command_addr[total_command_size/4] = (pad_size/4 - 1) << 16;
+    dword_command_addr[0] = (pad_size/4 - 1) << 16;
   }
 
   ReleaseWriteAddress(curr_index, total_command_size + pad_size);
