--- ./test/CMakeLists.txt~	2023-01-29 17:01:25.000000000 +0000
+++ ./test/CMakeLists.txt	2023-01-29 17:06:10.689999064 +0000
@@ -27,9 +27,8 @@
 set(CMAKE_EXECUTABLE_RUNTIME_HIP_FLAG_SEP ${CMAKE_SHARED_LIBRARY_RUNTIME_CXX_FLAG_SEP})
 set(CMAKE_EXECUTABLE_RPATH_LINK_HIP_FLAG ${CMAKE_SHARED_LIBRARY_RPATH_LINK_CXX_FLAG})
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${ROCM_PATH}/lib/cmake/hip")
 set(CMAKE_HIP_ARCHITECTURES OFF)
-find_package(HIP REQUIRED MODULE)
+find_package(HIP REQUIRED)
 
 find_package(Clang REQUIRED CONFIG
              PATHS "${ROCM_PATH}"
--- ./test/CMakeLists.txt~	2023-01-29 17:08:18.000000000 +0000
+++ ./test/CMakeLists.txt	2023-01-29 17:42:46.979873837 +0000
@@ -41,7 +41,7 @@
 
 ## Build MatrixTranspose
 set_source_files_properties(hip/MatrixTranspose.cpp PROPERTIES HIP_SOURCE_PROPERTY_FORMAT 1)
-hip_add_executable(MatrixTranspose hip/MatrixTranspose.cpp)
+HIP_ADD_EXECUTABLE(MatrixTranspose hip/MatrixTranspose.cpp)
 target_include_directories(MatrixTranspose PRIVATE ${PROJECT_SOURCE_DIR}/inc)
 target_link_libraries(MatrixTranspose PRIVATE roctracer roctx)
 add_dependencies(mytest MatrixTranspose)
@@ -49,7 +49,7 @@
 ## Build MatrixTranspose_test, MatrixTranspose_hipaact_test and MatrixTranspose_mgpu
 set_source_files_properties(app/MatrixTranspose_test.cpp PROPERTIES HIP_SOURCE_PROPERTY_FORMAT 1)
 function(build_matrix_transpose_test OUTPUT_FILE DEFINITIONS)
-  hip_add_executable(${OUTPUT_FILE} app/MatrixTranspose_test.cpp)
+  HIP_ADD_EXECUTABLE(${OUTPUT_FILE} app/MatrixTranspose_test.cpp)
   target_compile_definitions(${OUTPUT_FILE} PRIVATE ITERATIONS=100 HIP_TEST=1 ${DEFINITIONS})
   target_include_directories(${OUTPUT_FILE} PRIVATE ${PROJECT_SOURCE_DIR}/inc)
   target_link_libraries(${OUTPUT_FILE} PRIVATE roctracer roctx)
@@ -63,7 +63,7 @@
 ## Build MatrixTranspose MatrixTranspose_ctest
 add_custom_command(OUTPUT MatrixTranspose.c
   COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_CURRENT_SOURCE_DIR}/app/MatrixTranspose_test.cpp MatrixTranspose.c)
-hip_add_executable(MatrixTranspose_ctest MatrixTranspose.c)
+HIP_ADD_EXECUTABLE(MatrixTranspose_ctest MatrixTranspose.c)
 target_compile_definitions(MatrixTranspose_ctest PRIVATE HIP_TEST=0 __HIP_PLATFORM_HCC__)
 target_include_directories(MatrixTranspose_ctest PRIVATE ${PROJECT_SOURCE_DIR}/inc)
 target_link_libraries(MatrixTranspose_ctest PRIVATE roctracer roctx)
@@ -103,7 +103,7 @@
 
 ## Build the ROCTX test
 set_source_files_properties(app/roctx_test.cpp PROPERTIES HIP_SOURCE_PROPERTY_FORMAT 1)
-hip_add_executable(roctx_test app/roctx_test.cpp)
+HIP_ADD_EXECUTABLE(roctx_test app/roctx_test.cpp)
 target_link_libraries(roctx_test Threads::Threads roctx)
 add_dependencies(mytest roctx_test)
 
@@ -130,13 +130,13 @@
 
 ## Build the activity_and_callback test
 set_source_files_properties(directed/activity_and_callback.cpp PROPERTIES HIP_SOURCE_PROPERTY_FORMAT 1)
-hip_add_executable(activity_and_callback directed/activity_and_callback.cpp)
+HIP_ADD_EXECUTABLE(activity_and_callback directed/activity_and_callback.cpp)
 target_link_libraries(activity_and_callback roctracer)
 add_dependencies(mytest activity_and_callback)
 
 ## Build the multi_pool_activities test
 set_source_files_properties(directed/multi_pool_activities.cpp PROPERTIES HIP_SOURCE_PROPERTY_FORMAT 1)
-hip_add_executable(multi_pool_activities directed/multi_pool_activities.cpp)
+HIP_ADD_EXECUTABLE(multi_pool_activities directed/multi_pool_activities.cpp)
 target_link_libraries(multi_pool_activities roctracer)
 add_dependencies(mytest multi_pool_activities)
 
