--- ./src/comgr-objdump.cpp~	2022-12-08 19:58:50.000000000 +0000
+++ ./src/comgr-objdump.cpp	2022-12-08 19:59:53.613185220 +0000
@@ -2162,7 +2162,7 @@
 
   StringRef ClangASTContents;
   Expected<StringRef> ExpClangASTContents =
-      ClangASTSection.getValue().getContents();
+      ClangASTSection.value().getContents();
   if (ExpClangASTContents) {
     ClangASTContents = *ExpClangASTContents;
   } else {
@@ -2210,7 +2210,7 @@
 
   StringRef FaultMapContents;
   Expected<StringRef> ExpFaultMapContents =
-      FaultMapSection.getValue().getContents();
+      FaultMapSection.value().getContents();
   if (ExpFaultMapContents) {
     FaultMapContents = *ExpFaultMapContents;
   } else {
--- ./src/comgr-objdump.cpp~	2022-12-08 20:08:05.000000000 +0000
+++ ./src/comgr-objdump.cpp	2022-12-08 20:09:23.418581812 +0000
@@ -2203,7 +2203,7 @@
 
   OutS << "FaultMap table:\n";
 
-  if (!FaultMapSection.hasValue()) {
+  if (!FaultMapSection.has_value()) {
     OutS << "<not found>\n";
     return;
   }
--- ./src/comgr-compiler.cpp~	2022-12-08 20:22:54.000000000 +0000
+++ ./src/comgr-compiler.cpp	2022-12-08 20:23:08.046039787 +0000
@@ -245,12 +245,13 @@
                                      OPT_compress_debug_sections_EQ)) {
     if (A->getOption().getID() == OPT_compress_debug_sections) {
       // TODO: be more clever about the compression type auto-detection
-      Opts.CompressDebugSections = llvm::DebugCompressionType::Z;
+      Opts.CompressDebugSections = llvm::DebugCompressionType::Zlib;
     } else {
       Opts.CompressDebugSections =
           llvm::StringSwitch<llvm::DebugCompressionType>(A->getValue())
               .Case("none", llvm::DebugCompressionType::None)
-              .Case("zlib", llvm::DebugCompressionType::Z)
+              .Case("zlib", llvm::DebugCompressionType::Zlib)
+              .Case("zstd", llvm::DebugCompressionType::Zstd)
               .Default(llvm::DebugCompressionType::None);
     }
   }
