--- ./CMakeLists.txt.orig	2022-10-15 00:03:23.000000000 +0100
+++ ./CMakeLists.txt	2022-10-15 00:09:59.908248986 +0100
@@ -35,28 +35,9 @@
 
 # Link against LLVM and CLANG libraries
 target_link_libraries(hipify-clang PRIVATE
-    clangASTMatchers
-    clangFrontend
-    clangTooling
-    clangParse
-    clangSerialization
-    clangSema
-    clangEdit
-    clangFormat
-    clangLex
-    clangAnalysis
-    clangDriver
-    clangAST
-    clangToolingCore
-    clangRewrite
-    clangBasic
-    LLVMProfileData
-    LLVMSupport
-    LLVMMCParser
-    LLVMMC
-    LLVMBitReader
-    LLVMOption
-    LLVMCore)
+    clang-cpp
+    LLVM
+    LLVMSupport)
 
 if(LLVM_PACKAGE_VERSION VERSION_GREATER "6.0.1")
     target_link_libraries(hipify-clang PRIVATE clangToolingInclusions)
@@ -39,18 +19,6 @@
     LLVM
     LLVMSupport)
 
-if(LLVM_PACKAGE_VERSION VERSION_GREATER "6.0.1")
-    target_link_libraries(hipify-clang PRIVATE clangToolingInclusions)
-endif()
-
-if(LLVM_PACKAGE_VERSION VERSION_GREATER "9.0.1")
-    target_link_libraries(hipify-clang PRIVATE LLVMFrontendOpenMP)
-endif()
-
-if(LLVM_PACKAGE_VERSION VERSION_EQUAL "15.0.0" OR LLVM_PACKAGE_VERSION VERSION_GREATER "15.0.0")
-    target_link_libraries(hipify-clang PRIVATE LLVMWindowsDriver clangSupport)
-endif()
-
 if(MSVC)
     target_link_libraries(hipify-clang PRIVATE version)
     target_compile_options(hipify-clang PRIVATE /std:c++14 /Od /GR- /EHs- /EHc-)
